plugins/inputs/zfs/zfs_freebsd.go

```diff
@@ -106,6 +107,7 @@ func (z *Zfs) gatherDatasetStats(acc telegraf.Accumulator) (string, error) {
 		for _, line := range lines {
 			col := strings.Split(line, "\t")
 			if len(col) != len(properties) {
+				log.Println("The number of columns in the command result doesn't match the number of columns expected.")
```

Comment:
Not quite. You have to add the logger
```
 Log      telegraf.Logger `toml:"-"`
```
to your data structure and here use something like
```
z.Log.Warnf("Invalid number of columns for line: %s", line)
```

Reply:
You can mark this as resolved.